mmv1/third_party/terraform/services/composer/data_source_google_composer_user_workloads_secret.go.erb

```diff
@@ -0,0 +1,67 @@
+<% autogen_exception -%>
+package composer
+
+<% unless version == 'ga' -%>
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-google/google/tpgresource"
+	transport_tpg "github.com/hashicorp/terraform-provider-google/google/transport"
+)
+
+func DataSourceGoogleComposerUserWorkloadsSecret() *schema.Resource {
+	dsSchema := tpgresource.DatasourceSchemaFromResourceSchema(ResourceComposerUserWorkloadsSecret().Schema)
+
+	// Set 'Required' schema elements
+	tpgresource.AddRequiredFieldsToSchema(dsSchema, "environment", "name")
+
+	// Set 'Optional' schema elements
+	tpgresource.AddOptionalFieldsToSchema(dsSchema, "project", "region")
+
+	return &schema.Resource{
+		Read:   dataSourceGoogleComposerUserWorkloadsSecretRead,
+		Schema: dsSchema,
+	}
+}
+
+func dataSourceGoogleComposerUserWorkloadsSecretRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*transport_tpg.Config)
+
+	id, err := tpgresource.ReplaceVars(d, config, "projects/{{project}}/locations/{{region}}/environments/{{environment}}/userWorkloadsSecrets/{{name}}")
+	if err != nil {
+		return fmt.Errorf("Error constructing id: %s", err)
+	}
+	d.SetId(id)
+
+	// retrieve "data" in advance, because Read function won't do it.
+	userAgent, err := tpgresource.GenerateUserAgentString(d, config.UserAgent)
+	if err != nil {
+		return err
+	}
+
+	res, err := config.NewComposerClient(userAgent).Projects.Locations.Environments.UserWorkloadsSecrets.Get(id).Do()
```

Comment:
Normally data sources would correspond pretty closely to the Read function of the resource; I noticed that this line is present in the resource's Import function but not in Read. Why is that?

Reply:
What is the use case for this data source if the secrets are masked? Most similar data sources are useful because they provide access to the real secret value, which is then referenced from other resources. For example, see: https://registry.terraform.io/providers/hashicorp/google/latest/docs/data-sources/secret_manager_secret_version_access

Reply:
Only values in "data" filed's key-value pairs are masked. I think it would still be useful to manage this StringMap (update values for existing keys, add new key-value pairs and so on).

Reply:
This call is specifically to get the "data" field, but the values are all masked. Users won't be able to update the fields using the data source, since it's read-only. Could you elaborate?

Reply:
Right, I missed the fact that it is read-only. The remaining use case is reading keys of "data" field in a secret that is defined outside of Terraform.